libs/core/include/core/state_machine.hpp

```diff
@@ -293,7 +307,26 @@ char const *StateMachine<S>::GetStateName() const
 }
 
 /**
- * Determine if the runnable is ready to the run again
+ * Get the current string representation for the previous state
+ *
+ * @tparam S The state enum type
+ * @return The string representation of the current state if successful otherwise "Unknown"
+ */
+template <typename S>
+char const *StateMachine<S>::GetPrevStateName() const
+{
+  char const *text = "Unknown";
+
+  if (mapper_)
+  {
+    text = mapper_(previous_state());
+  }
+
+  return text;
```

Comment:
NP: this can be a single line.

Reply:
do you mean the whole function could be a ternary operator?

Reply:
Yes.

Reply:
I will probably leave it as is since it mirrors the function above it `GetStateName`